lib/fetch/headers.js

```diff
@@ -149,6 +150,60 @@ class HeadersList extends Array {
   }
 }
 
+class FilteredHeadersList extends HeadersList {
+  /** @type {(name: string) => boolean} */
+  filter = () => true;
```

Comment:
Needed `babel-eslint` for standard because it couldn't parse this properly. There's an issue on github with this solution I can pull up if you need.